--- a/Makefile
+++ b/Makefile
@@ -1354,8 +1354,13 @@
 
 # *** AUTODETECT ***
 ifeq ($(CONFIG_PLATFORM_AUTODETECT), y)
+
+ifdef EXT_EXTRA_CFLAGS
+EXTRA_CFLAGS += $(EXT_EXTRA_CFLAGS)
+else
 EXTRA_CFLAGS += -DCONFIG_LITTLE_ENDIAN
 EXTRA_CFLAGS += -DCONFIG_IOCTL_CFG80211 -DRTW_USE_CFG80211_STA_EVENT
+endif
 
 SUBARCH := $(shell uname -m)
 
@@ -1373,10 +1378,18 @@
 
 ARCH ?= $(SUBARCH)
 
+ifndef CROSS_COMPILE
 CROSS_COMPILE ?=
+endif
+ifndef KVER
 KVER ?= $(shell uname -r)
+endif
+ifndef KSRC
 KSRC := /lib/modules/$(KVER)/build
+endif
+ifndef MODDESTDIR
 MODDESTDIR := /lib/modules/$(KVER)/kernel/drivers/net/wireless/
+endif
 INSTALL_PREFIX :=
 STAGINGMODDIR := /lib/modules/$(KVER)/kernel/drivers/staging
 
